docs/dependencies.md

```diff
@@ -0,0 +1,213 @@
+# Dependencies
+
+## Core Spring Boot Dependencies
+
+- **spring-boot-starter-web**:  
+  Provides the essential components for building web applications with Spring MVC.
```

Comment:
We should move all Starter deps to a central `Spring Dependencies` category or move all dependencies to its concerned cross cutting concept aka. subcategory.